stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2495,38 +2482,62 @@ collections = [
 ]
 }%
 
+% for (Self, a_self, TypeParametersDecl, TypeParameters, AnyTypeParameters, Sequence, AnySequenceType) in collections:
+
 /// Header part of the native storage.
-internal struct _HashedContainerStorageHeader {
+internal struct _${Self}StorageHeader<${TypeParameters}> {
+
+%if Self == 'Set':
+  internal typealias Key = ${TypeParameters}
+  internal typealias Value = ${TypeParameters}
+%end
+
   internal init(capacity: Int) {
     self.capacity = capacity
   }
 
-  internal var capacity: Int
+  internal let capacity: Int
   internal var count: Int = 0
-  internal var maxLoadFactorInverse: Double =
+  internal let maxLoadFactorInverse: Double =
     _hashContainerDefaultMaxLoadFactorInverse
-}
 
-% for (Self, a_self, TypeParametersDecl, TypeParameters, AnyTypeParameters, Sequence, AnySequenceType) in collections:
+  // Placeholder values until the struct can be initialized.
```

Comment:
Yes, although I'm unsure what the UB implications for the empty singleton is if it doesn't initialize these at all. It *should* be guaranteed to never access them, if that matters. Just initialize them to point at 0x01?